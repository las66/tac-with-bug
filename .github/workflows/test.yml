name: Run test suite

on: pull_request

jobs:
  build:

    runs-on: ubuntu-latest
    
    services:
      postgres:
        image: postgres:latest
        env:
          POSTGRES_DB: tac        
          POSTGRES_PASSWORD: postgres
          POSTGRES_USER: postgres
        ports:
          - 5432:5432
        # Set health checks to wait until postgres has started
        options: >-
          --health-cmd pg_isready
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5

    steps:
    - uses: actions/checkout@v2
    - name: Use Node.js 16.x
      env: 
        HEROKU_API_KEY: ${{ secrets.HEROKU_API_KEY }}
      uses: actions/setup-node@v2
      with:
        node-version: 16.x
        cache: 'npm'
        cache-dependency-path: |
          server/package-lock.json
          client/package-lock.json
          shared/package-lock.json
    - name: Install shared modules
      run: |
        cd shared
        npm ci
    - name: Linting shared
      run: |
        cd shared
        npm run lint
    - name: Install server modules
      run: |
        cd server
        npm ci
    - name: Check linting and formatting of server
      run: |
        cd server
        npm run lint
        npm run prettier-check
    - name: Build server documentation
      run: |
        cd server
        npm run building
    - name: Run typescript build
      run: |
        cd server
        npm run tsc
    - name: Check Locale Files (All languages and identical keys)
      run: |
        cd server
        npm run checkLocales
    - name: Install client modules
      run: |
        cd client
        npm ci
    - name: Run client lint
      run: |
        cd client
        npm run lint
    - name: Run client tsc
      run: |
        cd client
        npm run tsc
    - name: Build Vue Files
      run: |
        cd client
        npm run build
    - name: Run server test
      run: |
        cd server
        npm run test -- --coverage=false
      env:
        paypal_Secret: ${{ secrets.PAYPAL_SECRET }}
