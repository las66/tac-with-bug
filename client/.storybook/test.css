@import url('../node_modules/primevue/resources/themes/saga-blue/theme.css') (prefers-color-scheme: light);
@import url('../node_modules/primevue/resources/themes/arya-blue/theme.css') (prefers-color-scheme: dark);
/*
:root {
  --tac-red: #ef3f23;
  --tac-text-color: #002b54;
}

html {
  padding: 0;
  margin: 0;
  height: 100%;
  width: 100%;
  background-color: var(--surface-b);
}

body {
  padding: 0;
  margin: 0;
  width: 100%;
  height: 100%;
  overflow: hidden;
}

@media (max-width: 600px) {
  .bodyNoscroll {
    height: 100%;
    overflow: hidden;
    bottom: 0;
    left: 0;
    position: fixed;
    right: 0;
    top: 0;
  }
}

#app {
  width: 100%;
  height: 100%;
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: var(--text-color);
}

.clickable {
  cursor: pointer;
}

* {
  scrollbar-width: thin;
  scrollbar-color: var(--primary-color) transparent;
}

*::-webkit-scrollbar {
  width: 12px;
}

*::-webkit-scrollbar-track {
  background: transparent;
}

*::-webkit-scrollbar-thumb {
  background-color: var(--primary-color);
  border-radius: 20px;
  border: 2px solid transparent;
  background-clip: padding-box;
}

@font-face {
  font-family: 'tacfontregular';
  src: url('./assets/TacFont/tacfont_semibold_1-webfont.woff2') format('woff2'), url('./assets/TacFont/tacfont_semibold_1-webfont.woff') format('woff');
  font-weight: normal;
  font-style: normal;
}

.tacLetters {
  user-select: none;
}

.tacLetters::first-letter {
  letter-spacing: -0.17em;
}

.twbLetters {
  font-family: 'tacfontregular', Monospace;
  letter-spacing: -0.17em;
  padding-right: 0.17em;
  user-select: none;
}

.twbLetters::first-letter {
  letter-spacing: -0.16em;
}

input:-webkit-autofill,
input:-webkit-autofill:hover,
input:-webkit-autofill:active,
input:-webkit-autofill:focus {
  filter: none;
  -webkit-text-fill-color: var(--text-color);
  box-shadow: 0 0 0 1000px rgba(15, 63, 81, 0) inset;
  transition: background-color 5000s ease-in-out 0s;
}

.floatingTextInput {
  margin-top: 30px !important;
}
*/
