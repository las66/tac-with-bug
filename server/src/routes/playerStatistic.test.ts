import { registerUserAndReturnCredentials, unregisterUser, User } from '../test/handleUserSockets';

describe('Platform PlayerStatistic Test Suite', () => {
    let userWithCredentials: User;

    beforeAll(async () => {
        userWithCredentials = await registerUserAndReturnCredentials()
    })

    afterAll(async () => {
        await unregisterUser(userWithCredentials)
    })

    test('Should return empty playerStats for new user', async () => {
        const response = await testAgent.get('/gameApi/profile/getPlayerStats/')
            .query({ username: userWithCredentials.username })
        expect(response.status).toBe(200)
        expect(response.body.history).toEqual([])
        expect(response.body.players.mostFrequent).toEqual('')
        expect(response.body.players.bestPartner).toEqual('')
        expect(response.body.players.worstEnemy).toEqual('')
        expect(response.body.table).toEqual([0, 0, 0, 0, 0, 0, 0])
        expect(response.body.subscriber).toEqual(false)
        expect(response.body.people).toEqual({})
        expect(response.body.hof).toEqual([])
        expect(response.body.userDescription).toEqual('')
        expect(response.body.gamesDistribution).toEqual({
            teamWon: 0,
            teamAborted: 0,
            won4: 0,
            lost4: 0,
            won6: 0,
            lost6: 0,
            aborted: 0,
            running: 0
        })
    })

    test('Should return empty userGraph for new user', async () => {
        const response = await testAgent.get('/gameApi/profile/userNetwork/')
            .query({ username: userWithCredentials.username })
        expect(response.status).toBe(200)
        expect(response.body.graph.nodes.length).toEqual(1)
        expect(response.body.graph.nodes[0].data.name).toBe(userWithCredentials.username)
        expect(response.body.graph.edges).toEqual([])
        expect(response.body.people).toEqual({})
    })

    test('Should return empty tournamentParticipations for new user', async () => {
        const response = await testAgent.get('/gameApi/profile/userTournamentParticipations/')
            .query({ username: userWithCredentials.username })
        expect(response.status).toBe(200)
        expect(response.body).toEqual([])
    })

    test.skip('Should return playerStats of Oskar', async () => {
        const response = await testAgent.get('/gameApi/profile/getPlayerStats/')
            .query({ username: 'Oskar' })
        expect(response.status).toBe(200)
    })

    test.skip('Should return userGraph of Oskar', async () => {
        const response = await testAgent.get('/gameApi/profile/userNetwork/')
            .query({ username: 'Oskar' })
        expect(response.status).toBe(200)
    })

    test.skip('Should return tournamentParticipations of Oskar', async () => {
        const response = await testAgent.get('/gameApi/profile/userTournamentParticipations/')
            .query({ username: 'Oskar' })
        expect(response.status).toBe(200)
        expect(response.body.length).toBeGreaterThan(0)
        expect(response.body[0].id).toBe(1)
        expect(response.body[0].title).toBe('March Madness Tournament')
        expect(response.body[0].date).toBe('2021-03-13T17:00:00.000Z')
        expect(response.body[0].team.name).toBe('Team Affenhaus')
        expect(response.body[0].team.players.sort()).toEqual(['Oskar', 'Sophia'])
        expect(response.body[0].team.playerids.sort()).toEqual([4, 7])
        expect(response.body[0].exitRound).toBe(4)
        expect(response.body[0].totalRounds).toBe(4)
        expect(response.body[0].placement).toBe(4)
    })

    test.todo('Test something more specific for Oskar stats')
})
