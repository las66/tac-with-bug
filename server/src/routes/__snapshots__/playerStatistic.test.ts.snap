// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Platform PlayerStatistic Test Suite Should return playerStats of existing player 1`] = `
Object {
  "gamesDistribution": Object {
    "aborted": 2,
    "lost4": 0,
    "lost6": 0,
    "running": 2,
    "teamAborted": 0,
    "teamWon": 2,
    "won4": 1,
    "won6": 0,
  },
  "history": Array [
    "coop",
    "coop",
    "won",
    "running",
    "running",
    "aborted",
    "aborted",
  ],
  "hof": Array [
    "verlag",
  ],
  "people": Object {
    "UserB": Array [
      0,
      0,
      1,
      1,
      3,
    ],
    "UserC": Array [
      1,
      1,
      0,
      0,
      3,
    ],
    "UserD": Array [
      0,
      0,
      1,
      1,
      3,
    ],
    "UserE": Array [
      0,
      0,
      0,
      0,
      1,
    ],
    "UserF": Array [
      0,
      0,
      0,
      0,
      1,
    ],
  },
  "players": Object {
    "bestPartner": "UserC",
    "mostFrequent": "UserB",
    "worstEnemy": "UserB",
  },
  "subscriber": false,
  "table": Array [
    100,
    48,
    92,
    85,
    74,
    89,
    60,
  ],
  "userDescription": "",
}
`;

exports[`Platform PlayerStatistic Test Suite Should return tournamentParticipations of existing player 1`] = `
Array [
  Object {
    "date": "2022-01-01T00:10:00.000Z",
    "exitRound": 2,
    "id": 1,
    "placement": 2,
    "team": Object {
      "name": "TeamA",
      "playerids": Array [
        1,
        2,
      ],
      "players": Array [
        "UserA",
        "UserB",
      ],
    },
    "title": "TournamentA",
    "totalRounds": 2,
  },
]
`;

exports[`Platform PlayerStatistic Test Suite Should return userGraph of existing player 1`] = `
Object {
  "graph": Object {
    "edges": Array [
      Object {
        "data": Object {
          "id": "e0",
          "source": "1",
          "target": "2",
          "together": false,
          "weight": 1,
        },
      },
      Object {
        "data": Object {
          "id": "e1",
          "source": "1",
          "target": "3",
          "together": true,
          "weight": 0.6666666666666666,
        },
      },
      Object {
        "data": Object {
          "id": "e2",
          "source": "1",
          "target": "4",
          "together": false,
          "weight": 0.6666666666666666,
        },
      },
      Object {
        "data": Object {
          "id": "e3",
          "source": "2",
          "target": "3",
          "together": false,
          "weight": 1,
        },
      },
      Object {
        "data": Object {
          "id": "e4",
          "source": "2",
          "target": "4",
          "together": true,
          "weight": 0.6666666666666666,
        },
      },
      Object {
        "data": Object {
          "id": "e5",
          "source": "3",
          "target": "4",
          "together": false,
          "weight": 1,
        },
      },
      Object {
        "data": Object {
          "id": "e6",
          "source": "1",
          "target": "3",
          "together": false,
          "weight": 0.3333333333333333,
        },
      },
      Object {
        "data": Object {
          "id": "e7",
          "source": "1",
          "target": "4",
          "together": true,
          "weight": 0.3333333333333333,
        },
      },
      Object {
        "data": Object {
          "id": "e8",
          "source": "1",
          "target": "5",
          "together": false,
          "weight": 0.3333333333333333,
        },
      },
      Object {
        "data": Object {
          "id": "e9",
          "source": "1",
          "target": "6",
          "together": false,
          "weight": 0.3333333333333333,
        },
      },
      Object {
        "data": Object {
          "id": "e10",
          "source": "2",
          "target": "4",
          "together": false,
          "weight": 0.3333333333333333,
        },
      },
      Object {
        "data": Object {
          "id": "e11",
          "source": "2",
          "target": "5",
          "together": true,
          "weight": 0.3333333333333333,
        },
      },
      Object {
        "data": Object {
          "id": "e12",
          "source": "2",
          "target": "6",
          "together": false,
          "weight": 0.3333333333333333,
        },
      },
      Object {
        "data": Object {
          "id": "e13",
          "source": "3",
          "target": "5",
          "together": false,
          "weight": 0.3333333333333333,
        },
      },
      Object {
        "data": Object {
          "id": "e14",
          "source": "3",
          "target": "6",
          "together": true,
          "weight": 0.3333333333333333,
        },
      },
      Object {
        "data": Object {
          "id": "e15",
          "source": "4",
          "target": "5",
          "together": false,
          "weight": 0.3333333333333333,
        },
      },
      Object {
        "data": Object {
          "id": "e16",
          "source": "4",
          "target": "6",
          "together": false,
          "weight": 0.3333333333333333,
        },
      },
      Object {
        "data": Object {
          "id": "e17",
          "source": "5",
          "target": "6",
          "together": false,
          "weight": 0.3333333333333333,
        },
      },
    ],
    "nodes": Array [
      Object {
        "data": Object {
          "id": "1",
          "idInt": 1,
          "name": "UserA",
          "score": 1,
        },
      },
      Object {
        "data": Object {
          "id": "2",
          "idInt": 2,
          "name": "UserB",
          "score": 1,
        },
      },
      Object {
        "data": Object {
          "id": "3",
          "idInt": 3,
          "name": "UserC",
          "score": 1,
        },
      },
      Object {
        "data": Object {
          "id": "4",
          "idInt": 4,
          "name": "UserD",
          "score": 1,
        },
      },
      Object {
        "data": Object {
          "id": "5",
          "idInt": 5,
          "name": "UserE",
          "score": 0.3333333333333333,
        },
      },
      Object {
        "data": Object {
          "id": "6",
          "idInt": 6,
          "name": "UserF",
          "score": 0.3333333333333333,
        },
      },
    ],
  },
  "people": Object {
    "UserB": Array [
      0,
      0,
      1,
      1,
      3,
    ],
    "UserC": Array [
      1,
      1,
      0,
      0,
      3,
    ],
    "UserD": Array [
      0,
      0,
      1,
      1,
      3,
    ],
    "UserE": Array [
      0,
      0,
      0,
      0,
      1,
    ],
    "UserF": Array [
      0,
      0,
      0,
      0,
      1,
    ],
  },
}
`;
